payments-core/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -80,6 +80,25 @@ internal class DefaultFlowController @Inject internal constructor(
      */
     private val lazyPaymentFlowResultProcessor: Lazy<PaymentFlowResultProcessor<out StripeIntent, StripeIntentResult<StripeIntent>>>
 ) : PaymentSheet.FlowController {
+    val paymentOptionActivityLauncher: ActivityResultLauncher<PaymentOptionContract.Args>
+    val googlePayActivityLauncher: ActivityResultLauncher<StripeGooglePayContract.Args>
+
+    init {
+        paymentController.updateLaunchersWithCallback(
+            activityResultCaller,
+            ::onPaymentFlowResult
+        )
```

Comment:
Thinking about this more, I think you'll also need to null out the references in `StripePaymentController` when the Activity is destroyed.

Maybe add something like this?


```
        lifecycleOwner.lifecycle.addObserver(
            object : LifecycleObserver {
                @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
                fun onDestroy() {
                    paymentController.unregisterActivityResultCaller()
                }
            }
        )
```

Reply:
nice catch, since there's no direct access to lifecycleOwner, added a `#unregisterLaunchers` function and called them in `DefaultFlowController#finalize` and `PaymentSheetActivity#onDestroy`